.github/workflows/tests-nightly.yml

```diff
@@ -204,13 +199,77 @@ jobs:
           CYPRESS_grepTags: ${{ env.GREP_TAGS }}
           STRIPE_SECRET_KEY: ${{ secrets.STRIPE_SECRET_KEY }}
           STRIPE_PUBLIC_KEY: ${{ secrets.STRIPE_PUBLIC_KEY }}
+          SPLIT: ${{ strategy.job-total }}
+          SPLIT_INDEX: ${{ strategy.job-index }}
         with:
           install: false
-          parallel: true
-          group: 'UI - Firefox'
-          record: true
-          tag: ${{ steps.get-env-uri.outputs.ENV_URI }},${{ env.GREP_TAGS }}
           browser: firefox
+      - name: Create reports dir
+        continue-on-error: true
+        if: ${{ ! cancelled() }}
+        run: npm run qa:create-artifacts-dirs
+      - name: Move screenshots into reports dir
+        continue-on-error: true
+        if: ${{ ! cancelled() }}
+        run: npm run qa:move-screenshots
+      - name: Upload reports
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: report-${{ strategy.job-index }}
+          path: ./cypress/reports
+          retention-days: 5
+          if-no-files-found: ignore
+  manual-triggered-test-complete:
```

Comment:
Instead of copy-pasting, you could consider using composite actions to reduce code duplication: https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

Reply:
TBH I already started PR with [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows). Looks like this should also do the trick. What do you think?

Reply:
Yes this can also work fine 👍